ports/esp32/modnetwork.c

```diff
@@ -275,11 +275,41 @@ STATIC mp_obj_t esp_disconnect(mp_obj_t self_in) {
 
 STATIC MP_DEFINE_CONST_FUN_OBJ_1(esp_disconnect_obj, esp_disconnect);
 
-STATIC mp_obj_t esp_status(mp_obj_t self_in) {
-    return mp_const_none;
+
+STATIC mp_obj_t esp_status(size_t n_args, const mp_obj_t *args) {
+    wlan_if_obj_t *self = MP_OBJ_TO_PTR(args[0]);
+    wifi_mode_t mode;
+    wifi_sta_list_t station_list;
+    wifi_sta_info_t* stations;
+    char sta_mac[17];
+
+    switch ((uintptr_t)args[1]) {
+        case (uintptr_t)MP_OBJ_NEW_QSTR(MP_QSTR_stations) :
+	    mp_printf(MP_PYTHON_PRINTER, "getting connected stations... \n");
+	    ESP_EXCEPTIONS(esp_wifi_get_mode(&mode));
+	    if ((mode & WIFI_MODE_AP)==0) {
+	        // we will only print stations in soft-AP mode, i.e. WIFI_MODE_AP
+	        mp_printf(MP_PYTHON_PRINTER, "Error: soft-AP mode not activated.\n");
+	    } else {
+		ESP_EXCEPTIONS(esp_wifi_ap_get_sta_list(&station_list));
+		stations = (wifi_sta_info_t*)(station_list.sta);
+		mp_printf(MP_PYTHON_PRINTER,"No. of connected stations : %d\n",station_list.num);
+		for (int i=0; i<station_list.num; i++) {
+		    
+		    sprintf(sta_mac,"%02x:%02x:%02x:%02x:%02x:%02x",stations[i].mac[0],stations[i].mac[1],stations[i].mac[2],stations[i].mac[3],stations[i].mac[4],stations[i].mac[5]);
+		    mp_printf(MP_PYTHON_PRINTER,"Station %d: %s\n",i,sta_mac);
```

Comment:
Instead of printing, try to create and return a list of tuples with `(mac, rssi)`.  See `mp_obj_new_list()` and `mp_obj_new_tuple()` (look at the implementation of `esp_scan()` for an example of the use of these functions).

Reply:
am thinking along the same line, but if we were to include additional functionality in future (e.g. network.status('rssi') should return the RSSI of the AP signal in STA mode, and when called with no parameter, it should describe the network link status), how could we make sure the interface can accommodate these future use cases?

Reply:
> how could we make sure the interface can accommodate these future use cases?

Have a look at how `esp_config()` does it (although that's a little more complex because it allows also to set parameters).  Basically you want to 1) first check if `n_args==1`, and if so then just return the link status (currently it can just return 0, to be implemented later); 2) if `n_args==2` then need to switch on the 2nd arg and act appropriately, which may mean that the code needs to check if it's in AP or STA mode to see if the request makes sense.